File: mobile/lib/models/assistant.dart
/// Assistant response model (`AssistantOut` from `POST /api/assistant`).
library;

class AssistantResponse {
  const AssistantResponse({
    required this.intent,
    required this.status,
    required this.message,
    this.data,
  });

  final String intent;
  final String status;
  final String message;
  final Map<String, dynamic>? data;

  bool get isOk => status == 'OK';
  bool get isInsufficientData => status == 'INSUFFICIENT_DATA';

  factory AssistantResponse.fromJson(Object? json) {
    final map = json as Map<String, dynamic>;
    return AssistantResponse(
      intent: map['intent'] as String? ?? '',
      status: map['status'] as String? ?? '',
      message: map['message'] as String? ?? '',
      data: map['data'] is Map<String, dynamic>
          ? Map<String, dynamic>.from(map['data'] as Map)
          : null,
    );
  }
}

File: config/core/schemas/assistant.py
"""
KisanAI OS
Assistant / Intent Router Schemas
"""

from typing import Optional

from pydantic import BaseModel, Field


class AssistantRequest(BaseModel):
    """A free-text farmer query routed to a stable intent.

    ``text`` is the farmer's question (Hindi / Hinglish / English).
    ``soil`` and ``disease`` are optional verified context that the
    client may attach so CROP_STATUS / advice can use real data when
    it is available on the device.
    """

    text: str = Field(..., min_length=1)
    soil: Optional[dict] = None
    disease: Optional[dict] = None


class AssistantOut(BaseModel):
    """Honest, structured assistant response.

    ``intent`` is the routed intent code. ``status`` is one of
    ``OK``, ``INSUFFICIENT_DATA`` or ``UNAVAILABLE``. ``message`` is
    the Hindi answer built only from verified data or honest pointers;
    ``data`` carries the structured sections (farm, crops, weather,
    soil, disease, advice) when available.
    """

    intent: str
    status: str = "OK"
    message: str = ""
    data: Optional[dict] = None
